[wip] ui: add sortable legend to metrics page

xinhaoz opened the pull request:
WIP

Fixes: #118645

xinhaoz committed:
ui: add sortable legend to metrics page

WIP

Fixes: #118645

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/127735)
<!-- Reviewable:end -->

xinhaoz commented:
Prototype - not really sure what to do with the old legend so it's hidden for now, but maybe we need both a minified version and this expanded, sortable version?
https://www.loom.com/share/fecbc0b6228d4c7ea6cedd2d1d3b7589?sid=245c7757-d8c9-4b57-958b-2785e52d327e

dhartunian committed:
ui: add hovering focus behavior to metric graphs

This PR enables a few new behaviors on the metrics graphs in DB
Console:
* Show individual lines on hover
* Show closest point to mouse pointer
* Revamped legend that takes up static amount of space and
scrolls
* Node series names no longer include the IP address for brevity

Additionally there are some minor visual tweaks in the legend like a
gray colored timestamp, and circle series markers instead of squares.

Part of CRDB-37094
Part of #118645
Epic: CRDB-37557

Release note (ui change): the user experience for metrics charges in
DB Console now has hover behavior that focuses on individual lines and
shows values under the mouse pointer.
